compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1610,6 +1629,13 @@ class Definitions {
   @tu lazy val Function2SpecializedReturnClasses: PerRun[collection.Set[Symbol]] =
     new PerRun(Function2SpecializedReturnTypes.map(_.symbol))
 
+  def isSpecializableTuple(base: Symbol, args: List[Type])(using Context): Boolean =
+    args.length <= 2 && base.isClass && TupleSpecializedClasses.exists(base.asClass.derivesFrom) && args.match
+      case List(x)    => Tuple1SpecializedParamClasses().contains(x.typeSymbol)
```

Comment:
Don't you have to use `x.widenDealias` here?

Reply:
Let me see if I can break it without.  What made you think of that?

Reply:
Usually it's suspicious to take `.typeSymbol` on an arbitrary `Type` before erasure. Singleton types, or intersection types that involve `& Int`, for example, may cause trouble here.